tests/core/math/test_vector3.h

```diff
@@ -144,6 +144,18 @@ TEST_CASE("[Vector3] Interpolation methods") {
 	CHECK_MESSAGE(
 			Vector3(1, 0, 0).move_toward(Vector3(10, 0, 0), 3) == Vector3(4, 0, 0),
 			"Vector3 move_toward should work as expected.");
+	CHECK_MESSAGE(
+			Vector3(1, 0, 0).rotate_toward(Vector3(0, 1, 0), Math_PI * 0.25) == Vector3(0.707107, 0.707107, 0),
```

Comment:
1. Use the `Math_SQRT12` constant for improved readability and precision.
2. You should use `is_equal_approx` here, because it's acceptable for the output to be off by a tiny amount.